Add link to ML model service (#2661)
viamrobotics/docs: docs/ml/_index.md

{{<imgproc src="/ml/training.png" class="alignright" resize="400x" declaredimensions=true alt="ML training">}}

> Machine Learning (ML) provides your machines with the ability to adjust its behavior based on models that recognize patterns or make predictions. 

Author: ```suggestion
Machine Learning (ML) provides your machines with the ability to adjust its behavior based on models that recognize patterns or make predictions.
```

Common use cases include:

- Object detection and classification which enable machines to detect people, animals, plants, or other objects with bounding boxes, and to perform actions when they are detected.
- Speech recognition, natural language processing, and speech synthesis, which enable machines to verbally communicate with us.

However, your machine can make use of machine learning with nearly any kind of data.

> To use machine learning (ML), you must deploy a [ML model service](/ml/deploy/) which runs a specified ML model.

Reviewer: ```suggestion
To use machine learning (ML), you must deploy an [ML model service](/ml/deploy/) which runs a specified ML model.
```

> Viam natively supports [TensorFlow Lite](https://www.tensorflow.org/lite) ML models as long as your models adhere to the [model requirements](/ml/deploy/#tflite_cpu-limitations).

Reviewer: ```suggestion
Viam natively supports [TensorFlow Lite](https://www.tensorflow.org/lite) ML models with the `tflite_cpu` ML model service, as long as your models adhere to the [model requirements](/ml/deploy/#tflite_cpu-limitations).
```
(opt) Not sure if necessary

Other model frameworks are supported with [modular resources](/ml/deploy/#modular-resources).

> Once you have deployed a model with the ML model service you can use it with the [vision service](/ml/vision) or other modular resources from the [registry](/registry) that make use of the ML model service.

Author: ```suggestion
Once you have deployed a model with the ML model service you can use it with the [vision service](/ml/vision/) or other modular resources from the [registry](/registry/) that make use of the ML model service.
```

{{<imgproc src="/ml/training.png" class="alignright" resize="400x" declaredimensions=true alt="ML training">}}

> Viam includes a built-in [machine learning (ML) model service](//ml/deploy/) which provides your machine with the ability to learn from data and adjust its behavior based on insights gathered from that data.

Reviewer: So with the model frameworks stuff i've been working on it looks like there will be more services like `onnx_cpu`, and there are already the modular services. Do you think it might be good to rephrase with that in mind as built-in makes it seem like just one?

Or is built-in not referring to `tflite_cpu` specifically but the type of service and would that still be accurate as the api type is built in?

Author: Yes - good point
```suggestion
Machine Learning (ML) provides your machines with the ability to adjust its behavior based on models that recognize patterns or make predictions. 
```

Reviewer: That works with the sentence below
